src/solc_0.7/common/BaseWithStorage/WithMetaTransactionReceiver.sol

```diff
@@ -0,0 +1,32 @@
+pragma solidity 0.7.1;
+
+import "./WithAdmin.sol";
+
+contract WithMetaTransactionReceiver is WithAdmin {
```

Comment:
I created a different porting of WithMetatransactuonReceiver 
I called it WithMetaTransaction and it support both GSN style metatx and Sandbox style one : 

https://github.com/thesandboxgame/sandbox-private-contracts/blob/88f1d8b49fb6e24368cea947a7f00bc400d0ec89/src/solc_0.7/common/BaseWithStorage/WithMetaTransaction.sol
What do you think of using this one instead ?
It changes slightly the way we deal with metatx but mostly have the same gas cost, see : https://github.com/thesandboxgame/sandbox-private-contracts/blob/88f1d8b49fb6e24368cea947a7f00bc400d0ec89/src/solc_0.7/catalyst/AssetUpgrader.sol#L150-L159

Reply:
It is actually already in hardhat branch, so I guess you should be able to reuse it already

Reply:
I saw it there. I was trying to follow your naming convention using the prefix "With", but seems unnecessary for me to add the extra contract, will use `WithMetaTransaction`